tests/units/Services/Template/Validators/ButtonTest.php

```diff
@@ -11,165 +11,146 @@
 
 class ButtonTest extends TestCase
 {
+    private $validator;
+    private $template;
+
+    protected function setUp() :void
+    {
+        parent::setUp();
+
+        // $this->withoutExceptionHandling();
+
+        $this->template = new Obj(['buttons' => [$this->mockedButton()]]);
+    }
 
     /** @test */
-    public function cannot_validate_without_mandatory_attribute()
+    public function cannot_validate_without_mandatory_attributes()
     {
-        $this->validate(
-            [
-                'buttons' => [
-                    []
-                ],
-            ],
-            false
-        );
+        $this->template->get('buttons')->first()->forget('type');
+
+        $this->expectException(TemplateException::class);
+
+        $this->expectExceptionMessage('The following attributes are mandatory for custom buttons: "type", "icon", "class"');
```

Comment:
I have a question here.
I agree with you expectException is not enough here, but I think expectExceptionMessage can be fragile here.
Can we use expectExceptionMessageRegExp or can we change the code and add a new exception for each validation?

Reply:
It's true, it is fragile. We should definitely add more exceptions, at least one for every validator type.